mlrun/api/crud/projects.py

```diff
@@ -286,22 +287,32 @@ async def _get_project_resources_counters(
             self._cache["project_resources_counters"]["ttl"] = ttl_time
         return self._cache["project_resources_counters"]["result"]
 
+    @staticmethod
+    def _list_pipelines(session):
+        return mlrun.api.crud.Pipelines().list_pipelines(
+            session, "*", format_=mlrun.api.schemas.PipelinesFormat.metadata_only
+        )
+
     async def _calculate_pipelines_counters(
         self,
     ) -> typing.Dict[str, int]:
-        def _list_pipelines(session):
-            return mlrun.api.crud.Pipelines().list_pipelines(
-                session, "*", format_=mlrun.api.schemas.PipelinesFormat.metadata_only
-            )
-
-        project_to_running_pipelines_count = collections.defaultdict(int)
+        project_to_running_pipelines_count = collections.defaultdict(lambda: 0)
         if not mlrun.mlconf.resolve_kfp_url():
+            # If KFP is not configured, return dict with 0 counters (no running pipelines)
             return project_to_running_pipelines_count
 
-        _, _, pipelines = await fastapi.concurrency.run_in_threadpool(
-            mlrun.api.db.session.run_function_with_new_db_session,
-            _list_pipelines,
-        )
+        try:
+            _, _, pipelines = await fastapi.concurrency.run_in_threadpool(
+                mlrun.api.db.session.run_function_with_new_db_session,
+                self._list_pipelines,
+            )
+        except Exception as exc:
```

Comment:
why not use `MLRunNotFoundError` or something more specific? we don't want to hide problems

Reply:
we don't know what will be the exact error, that is the reason for that PR overall, we want to be able to handle any error that we will experience from the KFP API and still return partial data